packages/component-library-react/src/HTMLHeading.tsx

```diff
@@ -0,0 +1,21 @@
+/**
+ * @license EUPL-1.2
+ * Copyright (c) 2024 Frameless
+ */
+import clsx from 'clsx';
+import { ForwardedRef, forwardRef, HTMLAttributes, PropsWithChildren } from 'react';
+
+export interface HTMLHeadingProps extends PropsWithChildren<HTMLAttributes<HTMLHeadingElement>> {
+  level: number;
+}
+
+export const HTMLHeading = forwardRef(
+  ({ level, className, ...restProps }: PropsWithChildren<HTMLHeadingProps>, ref: ForwardedRef<HTMLHeadingElement>) => {
+    const HeadingX =
+      level === 2 ? 'h2' : level === 3 ? 'h3' : level === 4 ? 'h4' : level === 5 ? 'h5' : level === 6 ? 'h6' : 'h1';
+
+    return <HeadingX className={clsx(className)} {...restProps} ref={ref} />;
+  },
+);
```

Comment:
Waarschijnlijk is `p` een betere fallback dan `h1`, in de meeste gevallen. Ik stel voor dat we de implementatie uit Strapi gebruiken:
https://github.com/frameless/strapi/tree/main/packages/ui/src/components/HTMLHeading

Daar zijn ook nog extra unit tests die waardevol zijn voor hier.